cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestManager.java

```diff
@@ -259,21 +273,36 @@ public void clear() throws Exception {
   @Override
   public void deployDatasetModule(Id.Namespace namespace, String moduleName,
                                   Class<? extends DatasetModule> datasetModule) throws Exception {
-    throw new UnsupportedOperationException();
+    datasetModuleClient.add(Id.DatasetModule.from(namespace, moduleName),
+                            datasetModule.getName(),
+                            createModuleJarFile(datasetModule));
+  }
+
+  private File createModuleJarFile(Class<?> cls) throws IOException {
+    String version = String.format("1.0.%d-SNAPSHOT", System.currentTimeMillis());
+    File moduleJarFile = new File(tmpFolder, String.format("%s-%s.jar", cls.getSimpleName(), version));
+    Location deploymentJar = AppJarHelper.createDeploymentJar(locationFactory, cls);
+    Files.copy(Locations.newInputSupplier(deploymentJar), moduleJarFile);
+    return moduleJarFile;
```

Comment:
Yea, the test case deploys the module before deploying the app.
The app requires the module is there.